src/sentry/integrations/github/integration.py

```diff
@@ -231,6 +237,55 @@ def get_client(self) -> GitHubBaseClient:
             raise IntegrationError("Organization Integration does not exist")
         return GitHubApiClient(integration=self.model, org_integration_id=self.org_integration.id)
 
+    # CredentialLeasableMixin methods
+
+    def get_maximum_lease_duration_seconds(self) -> int:
+        return 60 * 60  # Access tokens are valid for an hour by default.
+
+    def refresh_access_token_with_minimum_validity_time(
+        self, token_minimum_validity_time: timedelta
+    ) -> CredentialLease:
+        access_token_data = self.get_client().get_access_token(
+            token_minimum_validity_time=token_minimum_validity_time
+        )
+
+        assert access_token_data is not None, "Expected Integration to have an access token"
+
+        return self._credential_lease_from_model()
+
+    def force_refresh_access_token(self) -> CredentialLease:
+        raise NotImplementedError("Force refresh access token is not supported for GitHub")
+
+    def get_active_access_token(self) -> CredentialLease:
```

Comment:
Calling the GH API will refresh the access token if it needs to be refreshed, is that intended here? What do the refresh functions do differently if we are getting the most active refresh token and creating the credential lease here?

Reply:
So they do 2 things differently:
1. They hoist _some_ of the integration mutation logic up into the installation class (though not much for GH yet. There will be a follow-up PR to clean this up later)
2. Providing some minimum lifetime to callers when possible. Handing a token over that's about to expire isn't a great developer experience, though this approach increases the possibility that a token is handed out then revoked by another downstream service requesting a token with a longer validity window. Open to ideas on this one.